oktapam/client/secret_folder.go

```diff
@@ -0,0 +1,309 @@
+package client
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/atko-pam/pam-sdk-go/client/pam"
+	"github.com/okta/terraform-provider-oktapam/oktapam/constants/attributes"
+	"github.com/okta/terraform-provider-oktapam/oktapam/logging"
+	"github.com/tomnomnom/linkheader"
+)
+
+const secretFolderSeparator = "/"
+
+// SecretFolder represents a secret folder entity.  We use this instead of the PAM SDK response since
+// there are multiple API response types which are used for secret folders and this provides a common
+// structure that can be used
+type SecretFolder struct {
+	Name            *string `json:"name"`
+	ID              *string `json:"id,omitempty"`
+	Description     *string `json:"description,omitempty"`
+	ParentFolderID  *string `json:"parent_folder_id,omitempty"`
+	ResourceGroupID *string `json:"-"`
+	ProjectID       *string `json:"-"`
+}
+
+func (s SecretFolder) ToResourceMap() map[string]any {
+	m := make(map[string]any, 8)
+
+	if s.Name != nil {
+		m[attributes.Name] = *s.Name
+	}
+	if s.ID != nil {
+		m[attributes.ID] = *s.ID
+	}
+	if s.Description != nil {
+		m[attributes.Description] = *s.Description
+	}
+	if s.ParentFolderID != nil {
+		m[attributes.ParentFolder] = *s.ParentFolderID
+	}
+
+	return m
+}
+
+func ListTopLevelSecretFolders(ctx context.Context, sdkClient SDKClientWrapper) ([]SecretFolder, error) {
+	var list []SecretFolder
+	request := sdkClient.SDKClient.SecretsAPI.ListTopLevelSecretFoldersForUser(ctx, sdkClient.Team)
+
+	for {
+		resp, httpResp, callErr := request.Execute()
```

Comment:
I think we can take out the logic to handle link headers in a common method. We can convert API response outside that common method. If you want, we do it in separate PR too.